src/java.base/share/classes/sun/security/rsa/RSACore.java

```diff
@@ -53,10 +57,12 @@
     // globally enable/disable use of blinding
     private static final boolean ENABLE_BLINDING = true;
 
-    // cache for blinding parameters. Map<BigInteger, BlindingParameters>
-    // use a weak hashmap so that cached values are automatically cleared
-    // when the modulus is GC'ed
-    private static final Map<BigInteger, BlindingParameters>
+    // cache for blinding parameters. Map<BigInteger,
+    // ConcurrentLinkedQueue<BlindingParameters>> use a weak hashmap so that,
+    // cached values are automatically cleared when the modulus is GC'ed.
+    // Multiple BlindingParameters can be queued during times of heavy load,
+    // like performance testing.
+    private static final Map<BigInteger, ConcurrentLinkedQueue<BlindingParameters>>
                 blindingCache = new WeakHashMap<>();
```

Comment:
I'd use a synchronizedMap here now

Reply:
Using a synchronizedMap would only let one thread access the hashmap at a time, which is what I'm trying to reduce.  Get ops do not need to be locked and put ops don't either because the values are objects of random data, losing one is not significant to the operation.

Reply:
According to SO, unsynchronized writes to a map can lead to lock up, see here:
https://stackoverflow.com/a/1003237
Or here:
https://mailinator.blogspot.com/2009/06/beautiful-race-condition.html?m=1
What's the performance impact of adding synchronization on the hash map only?

Reply:
Ok, thanks for pointing that out, I didn't realize that limitation.  However using synchronizedMap is overkill as it locks every operation.  As I read it, only structural changes like adding and removing keys from the map causes problems.  Getting a value from the map does not.  Therefore I would only need to synchronize the putIfAbsent() at 449, which is pretty minor.  The expunging of the map is already synchronized in WeakHashMap.

The original code was half the performance of this one.

Reply:
If you synchronize `putIfAbsent` but not `get` you'll still get the possible infinite loop described in the SO post.

Reply:
Expunging entries is synchronized over an internal data structure, so if you synchronize puts only, nothing prevents expunge (called from get) from running in parallel with put.
The original code performed big integer exponentiation inside a synchronized block, your code doesn't. I expect substantial performance gains even with synchronization around hash map access, this is why I asked about performance.

Reply:
I finally see that get() is affected by the locking.  Expunge seems to not lock the able which could cause multiple threads get ops to corrupt the map.  I'll put a ReentrantLock around the put & get.  I prefer to not use synchronized() blocks as ReentrantLock maybe better after Project Loom

Sure, the performance was largely affected by the math ops in the old code.  I wanted to eliminate as much of the performance decelerators as possible